include/datatypes/impl/chunk_edge_subview.hpp

```diff
@@ -0,0 +1,207 @@
+#pragma once
+
+#include "datatypes/point_view.hpp"
+#include "enumerations/dimension.hpp"
+#include <Kokkos_Core.hpp>
+
+namespace specfem::datatype::impl {
+
+/**
+ * @brief Subview for accessing vector components within a chunk element
+ *
+ * This class provides a convenient interface for accessing vector data
+ * components at a specific index within a larger multi-dimensional view.
+ * It simplifies operations by abstracting the full indexing requirements.
+ *
+ * @tparam ViewType The type of the parent view this subview accesses
+ */
+template <typename ViewType> struct VectorChunkEdgeSubview {
```

Comment:
I think this needs an SFINAE to check if the constructing passed ViewType is a edge view type. 
```suggestion
template <typename ViewType, typename std::enable_if_t<specfem::data_access::is_chunk_edge<ViewType>, int = 0>> struct VectorChunkEdgeSubview {
```

Reply:
I'm not sure if SFINAE is necessary because it is in `impl` and can only be instantiated by `ChunkEdgeViewType`

Reply:
Okay, that makes sense.